### DISCUSSION TITLE
How to build a libvips.so.42 file with all dependencies included?

### DISCUSSION THREAD

**Discussion by fredprodibi** [libvips/libvips#3426] [Q&A]
## Questions

I m trying to use a custom build of libvips to use in libvips.NET that include jxl and heif.

I manage to build the code in docker but it produces a small libvips.so.42 and the two modules vips-heif.so and vips-jxl.so

How can I get a libvips.so.42 like libvips.NET does, but not only for the web formats, but also with jxl and heif included.

I use this command to setup the build
`meson build --libdir=lib --buildtype=release -Ddeprecated=false -Dopenslide=disabled -Dpdfium=disabled -Dpoppler=disabled -Dpoppler-module=disabled -Drsvg=disabled -Dopenslide-module=disabled -Dmatio=disabled -Dopenexr=disabled`

if I add --default-library static no .so files are produced.

Any help to build a version that I can use with libvips.NET that include all the web formats, + JPEG2000, + TIFF, + JXL and HEIC/AVIF would help.

Thank you

my docker image is a Debian GNU/Linux 11 (bullseye)
the dependency config is:
vips 8.14.2

  Build options
    enable debug                      : NO
    enable deprecated                 : NO
    enable modules                    : YES
    enable gtk-doc                    : NO
    enable doxygen                    : NO
    enable introspection              : YES
    enable examples                   : YES
    enable cplusplus                  : YES
    enable RAD load/save              : YES
    enable Analyze7 load/save         : YES
    enable PPM load/save              : YES
    enable GIF load                   : YES

  Optional external packages
    use fftw for FFTs                 : YES
    accelerate loops with ORC         : YES
    ICC profile support with lcms     : NO
    zlib                              : YES
    text rendering with pangocairo    : YES
    font file support with fontconfig : YES
    EXIF metadata support with libexif: YES

  External image format libraries
    JPEG load/save with libjpeg       : YES
    JXL load/save with libjxl         : YES (dynamic module: YES)
    JPEG2000 load/save with OpenJPEG  : YES
    PNG load/save with libspng        : NO
    PNG load/save with libpng         : YES
    selected quantisation package     : imagequant
    TIFF load/save with libtiff       : YES
    image pyramid save with libgsf    : YES
    HEIC/AVIF load/save with libheif  : YES (dynamic module: YES)
    WebP load/save with libwebp       : YES
    PDF load with PDFium              : NO
    PDF load with poppler-glib        : NO (dynamic module: NO)
    SVG load with librsvg             : NO
    EXR load with OpenEXR             : NO
    OpenSlide load                    : NO (dynamic module: NO)
    Matlab load with libmatio         : NO
    NIfTI load/save with niftiio      : NO
    FITS load/save with cfitsio       : YES
    GIF save with cgif                : NO
    selected Magick package           : none (dynamic module: NO)
    Magick API version                : none
    Magick load                       : NO
    Magick save                       : NO

  User defined options
    buildtype                         : release
    libdir                            : lib
    deprecated                        : false
    matio                             : disabled
    openexr                           : disabled
    openslide                         : disabled
    openslide-module                  : disabled
    pdfium                            : disabled
    poppler                           : disabled
    poppler-module                    : disabled
    rsvg                              : disabled

**Comment by kleisauke** [comment]
Meson prefers to link against shared libraries by default, one can use the `--prefer-static` option to try static linking before shared linking. The `-Dmodules=disabled` option could be used if you want to disable the support and build of dynamic modules.

The build script at https://github.com/kleisauke/libvips-packaging/blob/main/build/posix.sh can be used as a reference to compile libvips and its dependencies as a single, statically-linked shared library.

**Reply by wikke** [reply]
it just works. with libjxl still dynamic. So i just disable this feature. Everything works well
